### DISCUSSION TITLE
Back-relation expand but for a specific user Id

### DISCUSSION THREAD

**Discussion by MuslemRahimi** [pocketbase/pocketbase#3728] [Q&A]
Hi,

I have a weird case where i want to back expand

```
 output = await locals.pb.collection('posts').getList(1, 10, {
                expand: 'user, comments(post), alreadyVoted(post)',
            }) ;
```
where my output for the "alreadyVoted" collections is
```
console.log(output.items[0]?.expand['alreadyVoted(post)'])

[
  {
    collectionId: 'ofr9fkr5a3zpz91',
    collectionName: 'alreadyVoted',
    created: '2023-11-09 16:53:57.375Z',
    id: 'qe0us8dkd3hsbn4',
    post: 'rr98l6e1omwap1i',
    type: 'neutral',
    updated: '2023-11-09 19:16:26.408Z',
    user: 'db5s41oatgoeh0q'
  },
  {
    collectionId: 'ofr9fkr5a3zpz91',
    collectionName: 'alreadyVoted',
    created: '2023-11-09 19:08:56.577Z',
    id: '6d767bov0mx25te',
    post: 'rr98l6e1omwap1i',
    type: 'upvote',
    updated: '2023-11-09 19:08:56.577Z',
    user: 'zisfpgt90dajlmh'
  }
]

```

Here is my problem. I dont want to expand to get all user data for this collection but rather only for the userId = 'db5s41oatgoeh0q'.

Is there a way to do this elegantly?

**Comment by ganigeorgiev** [comment]
You can try specifying the `fields` parameter:

```js
output = await locals.pb.collection('posts').getList(1, 10, {
    expand: 'user, comments(post), alreadyVoted(post)',
    fields: "*,expand.alreadyVoted(post).user"
});
```
_(The `fields` format must follow the record json response structure.)_

**Reply by ganigeorgiev** [reply]
On second read I actually may misunderstood your question.

If you need to apply filter/sort/etc. on the `expand` result, then this is not supported and you'll have to send a separate request to the related collection.

`expand` expands all relations that the current request client has access to and that are assigned to the main record(s).
